internal/voting/voting.go

```diff
@@ -56,9 +56,50 @@ func initializeVoteStore(ctx context.Context, db sql.TxMaker) error {
 		}
 	}
 
+	err = validateResolutionTypes(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("failed to validate resolution types: %w", err)
+	}
+
 	return tx.Commit(ctx)
 }
 
+// validateResolutionTypes verifies if the resolutions stored in the vote store
+// are the same as the resolutions registered. Any mismatch at the startup would
+// indicate unplanned consensus changes and should be treated as a critical error.
```

Comment:
`validateResolutionTypes` is called *after* a loop around `createResolutionType`, so I believe only the length check could ever hit as it is currently used.  Did we want to detect resolutions an existing table that are not in the registered resolutions map from `ListResolutions`?